test/fixtures.py

```diff
@@ -202,7 +202,13 @@ class AddOneAggregate(AggregateExpFragment):
     def build_fragment(self) -> None:
         self.setattr_fragment("a", AddOneFragment)
         self.setattr_fragment("b", AddOneFragment)
-        return super().build_fragment([self.a, self.b])
+
+        self.setattr_result("sum", FloatChannel)
+
+        def push_sum():
+            self.sum.push(self.a.value.get() + 1 + self.b.value.get() + 1)
```

Comment:
I would love this to be `self.sum.push(self.a.result.get_last() + self.b.result.get_last())` but:
* currently the `get_last` is deliberately kernel only
* we could add a sink-based interface that works by absorbing results...but then that's host-only so we can't run the test in a portable fashion
* I also considered having `AddOneFragment` store a `last_value` attribute and store the result there as well, but that also seems like a pretty miserable hack. We're essentially saying to users "getting the last value pushed to a result is a legitimate thing to do, but we won't support it so you have to manually duplicate result channels and handle all the accompanying synchronization issues yourself"

Reply:
> We're essentially saying to users "getting the last value pushed to a result is a legitimate thing to do, but we won't support it so you have to manually duplicate result channels and handle all the accompanying synchronization issues yourself"

This is still preferrable to adding a half-baked implementation with unclear semantics to ndscan. "Magic" solutions that work only most of the time with unclear semantics (i.e. leaky abstractions) are much worse than just requiring users to write some straightforward Python code.